gateway-topology-simple/src/main/java/org/apache/knox/gateway/topology/simple/SimpleDescriptorHandler.java

```diff
@@ -195,6 +200,47 @@
                                 gws);
     }
 
+    /**
+     * Determine whether discovery should be performed for the specified descriptor.
+     *
+     * @param desc A SimpleDescriptor
+     * @return true, if discovery should be performed for the descriptor; Otherwise, false.
+     */
+    private static boolean shouldPerformDiscovery(final SimpleDescriptor desc) {
+
+        // If there is a discovery type specified, then discovery should be performed
+        final String discoveryType = desc.getDiscoveryType();
+        if (discoveryType != null && !discoveryType.isEmpty()) {
```

Comment:
Instead of relying on all the properties shouldn't we just rely on discoveryType? less confusion with if else logic and also while defining the descriptor.

Reply:
Discovery type has been optional since the beginning, and defaults to Ambari when omitted. I didn't want to break this by adding a requirement that discovery type be specified. What I'm saying here is that any discovery-related properties in a descriptor signal that discovery is expected, and the rest carries on as it has prior to this change.

Reply:
Okay, I see.